src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/util/FileUtil.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.app.impl.cloudsdk.internal.util;
+
+import com.google.common.base.Preconditions;
+
+import java.io.IOException;
+import java.nio.file.CopyOption;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.AclFileAttributeView;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.PosixFileAttributeView;
+import java.nio.file.attribute.PosixFileAttributes;
+
+/**
+ * Internal file utilities.
+ */
+public class FileUtil {
+
+  /**
+   * We really want to preserve file attributes and permissions on linux and windows.
+   *
+   * @param source an existing source directory to copy from
+   * @param destination an existing destination directory to copy to
+   */
+  public static void copyDirectory(final Path source, final Path destination) throws IOException {
```

Comment:
Well partially because we don't want to include that dependency. Secondly because I don't think it preserves attributes/permissions, but that might have changed, going to have another look at their source.